lib/ansible/runner/credential/openstack_credential.rb

```diff
@@ -0,0 +1,44 @@
+module Ansible
+  class Runner
+    class OpenstackCredential < Credential
+      def self.auth_type
+        "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::OpenstackCredential"
+      end
+
+      # Modeled off of openstack injectors for awx:
+      #
+      #   https://github.com/ansible/awx/blob/1242ee2b/awx/main/models/credential/injectors.py#L70-L96
+      #
+      def env_vars
+        write_credentials_file
+        { "OS_CLIENT_CONFIG_FILE" => os_credentials_file }
+      end
+
+      def write_credentials_file
```

Comment:
Same here. I think this should either be a private method or be implemented as `#write_password_file` and not called from `#env_vars`.

Reply:
So that is where I just made an executive decision and didn't overload `#write_password_file` since that is specifically designed to write `env/passwords`, and I didn't want to confuse the two.

That said, I also thought it should be a similar interface to `#write_password_file`, which is public.  That said, I am wondering if the interface maybe should change for how we are writing files:

https://github.com/ManageIQ/manageiq/pull/18968/files#diff-259e0b014fb9528a44a195b1ba3025a4R266

And maybe in `lib/ansible/runner/credential.rb` there should be a `write_env_files` method instead, that can be overwritten by the base class (and we make `write_password_file` `private` as well).  It would also allow for what we were thinking in this case:

https://github.com/ManageIQ/manageiq/pull/18968/files#r303051347

To log when we write a file, so it is clear what credential type created what, and if something does get clobbered (say an `ScmCredntial` clobbers the `MachineCredential`'s `env/password`), we would be able to see the order in the logs.

Reply:
I could also create a base class `CloudCrendential` for this to share some of this logic, but wasn't sure that was worth it.

Reply:
Nah, maybe if we just make it more generic. I never intended the method to care about the contents of the files. So would it make things more clear if we renamed it everywhere to `write_config_files` or even just make this method name the one in the base class?